### DISCUSSION TITLE
script for N-glycan analysis

### DISCUSSION THREAD

**Discussion by NasserMS2** [mobiusklein/ms_deisotope#31] [General]
Hi,

Im working on analyzing small molecules (N-glycnas) in mzML format. I'm trying to deconvolute them and then use my script to annotate them. I want to use deconv function from your script but not sure which one works best for molecules like glycans. I will appreciate if you provide me the code using ms_deisotope functions or modules to deconvolute my mzml containing m/z vs intensity and return the corresponding masses. if we can have output mass range and also input m/z range with input charge rage, would be great. I really need it. I tried several scripts for deconv but those are not good for small molecules.

**Comment by mobiusklein** [comment]
The best parameters will depend upon the type of data you have. `ms_deisotope` was written in part to support my own work on released *N*-glycan LC-MS and *N*-glycoproteomics.

If you just want to deisotope and charge state deconvolve an entire LC-MS file, you can use the command line tool `ms-deisotope` with the following options:
```bash
ms-deisotope -g 1 -a glycan -an glycan -m 2 -t 10 -tn 0 -b 1 $INPUT_FILE $OUTPUT_FILE
```
Where `-g 1` enables MS1 signal averaging, smoothing noisy data, `-a glycan` and `-an glycan` specify that the analytes will follow the glycan isotopic model, `-m 2` indicates that up to two missing isotopic peaks will be tolerated for MS1 fits, not uncommon for large molecules, `-t 10` specifies a fairly relaxed minimum fit quality for MS1 spectra that discards spurious fits but doesn't throw away everything, and `-tn 0` indicates that if there are MS2 spectra that it will tolerate even partial fits. The `-b 1` option does some additional background noise reduction, but depending upon your instrument you may want to set this to 0 or raise it to 5 if you have a lot of background noise, e.g. TOF or FTICR data.

This produces a new mzML file with isotopic peaks removed and an associated charge array for each peak. The fitted isotopic patterns are still stored in a separate array that `ms_deisotope` can unpack for you with `ms_deisotope.output.ProcessedMzMLLoader`.

If you need to do it programmatically, I can provide an example for that as well.

**Comment by NasserMS2** [comment]
Hi,

thanks for the quick reply. if you give me an example, it would be great.
